### PR TITLE
Camera view/fly-to bounding sphere

### PR DESCRIPTION
**Pull request by bagnell**
NOTE: This is a PR into the `entity-zoom` branch.

Modifies the `viewBoundingSphere` and `flyToBoundingSphere` functions so that the whole bounding sphere is visible. Also adds an optional offset.

### CODE REVIEW

**Comment by mramato** [comment]
Awesome.  Thanks @bagnell.  One issue I've found so far is that if you load the CZML example and double click on a satellite, you crash.  Probably something in EntityView.

File: Source/Scene/Camera.js [hunk 1]
```diff
@@ -2451,37 +2451,96 @@ define([
...
         }
         //>>includeEnd('debug');
 
-        var transform = Transforms.eastNorthUpToFixedFrame(boundingSphere.center);
-        this.transform = transform;
-        var r = 2.0 * Math.max(boundingSphere.radius, this.frustum.near);
-        this.lookAt(new Cartesian3(0, -r, r), Cartesian3.ZERO, Cartesian3.UNIT_Z);
-        this.setTransform(Matrix4.IDENTITY);
+        if (this._mode === SceneMode.MORPHING) {
+            return;
```
Comment (mramato) [inline Source/Scene/Camera.js]: This should probably throw a `DeveloperError` instead of simply returning.

**Comment by mramato** [comment]
Also, not specific to this PR, but something I just noticed is that the HomeButtonViewModel code is still using `direction` and `up` for flyTo, which results in messages in the console every time you fly home.

**Comment by bagnell** [comment]
>... the HomeButtonViewModel code is still using direction and up for flyTo ...

Thanks. I thought I caught them all.

This is ready for another review.

**Comment by mramato** [comment]
Okay, I submitted one minor tweak that was causing issues when the stop tracking button gets pressed.  However, I also found a few more issues:

1. Double click on any of the markers in the [simplestyle](http://localhost:8080/Apps/Sandcastle/index.html?src=GeoJSON%20simplestyle.html&label=DataSources) demo and you get a black screen followed by a crash as soon as you try and zoom home. Maybe similar to the last problem when boundingSphere radius is 0?

2. `createMorphHandler` in SceneTransitioner.js is triggering the `mouseMoveOnDocument parameter to the ScreenSpaceEventHandler constructor was deprecated in Cesium 1.6` warning.

**Comment by bagnell** [comment]
@mramato Alright, I think this is ready now.

**Comment by mramato** [comment]
I hate to say it, but there's a new problem.  The camera in the 3D model demo is now way farther than it used to be. I'm not sure why and I think it was good before your last commit.

**Comment by mramato** [comment]
Thanks @bagnell

**Pull request merged**